packages/block-library/src/search/index.php

```diff
@@ -64,23 +63,21 @@ function render_block_core_search( $attributes ) {
 		);
 	}
 
-	if ( $show_input ) {
-		$input_classes = array( 'wp-block-search__input' );
-		if ( ! $is_button_inside && ! empty( $border_color_classes ) ) {
-			$input_classes[] = $border_color_classes;
-		}
-		if ( ! empty( $typography_classes ) ) {
-			$input_classes[] = $typography_classes;
-		}
-		$input_markup = sprintf(
-			'<input type="search" id="%s" class="%s" name="s" value="%s" placeholder="%s" %s required />',
-			$input_id,
-			esc_attr( implode( ' ', $input_classes ) ),
-			get_search_query(),
-			esc_attr( $attributes['placeholder'] ),
-			$inline_styles['input']
-		);
+	$input_classes = array( 'wp-block-search__input' );
+	if ( ! $is_button_inside && ! empty( $border_color_classes ) ) {
+		$input_classes[] = $border_color_classes;
+	}
+	if ( ! empty( $typography_classes ) ) {
+		$input_classes[] = $typography_classes;
 	}
+	$input_markup = sprintf(
```

Comment:
It would be nice to do this using the `WP_HTML_Tag_Processor`. There's an example of this in https://github.com/WordPress/gutenberg/pull/49212/files. Not necessary to get his merged, but a nice improvement.

Reply:
Nice, happy to try that in a follow up.